lib/semantics/resolve-names.cpp

```diff
@@ -1226,6 +1232,18 @@ class OmpAttributeVisitor {
     auto it{GetContext().objectWithDSA.find(&symbol)};
     return it != GetContext().objectWithDSA.end();
   }
+  void SetContextAssociatedLoopLevel(std::size_t level) {
+    GetContext().associatedLoopLevel = level;
+  }
+  std::size_t GetAssociatedLoopLevelFromClauses(const parser::OmpClauseList &);
+
+  template<typename T>
+  std::optional<std::int64_t> GetClauseIntValue(const T &x) {
```

Comment:
Is there a similar routine somewhere else that does this?

Reply:
I searched around and didn't find one.

Reply:
Why not `ToInt64(EvaluateIntExpr(x))`?

Reply:
Yes, it works too. I can remove this template function then.